### PR TITLE
Add Get user by query

### PR DESCRIPTION
**Pull request by Toa741**

### CODE REVIEW

File: NGitLab/Models/UserQuery.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿namespace NGitLab
+{
+    public class UserQuery
+    {
+        /// <summary>
+        /// Specifies how many record per paging (Gitlab supports a maximum of 100 projects and defaults to 20).
```
Comment (louis-z) [inline NGitLab/Models/UserQuery.cs]:
```suggestion
        /// Specifies how many records per page (GitLab supports a maximum of 100 items per page and defaults to 20).
```

File: NGitLab/Models/UserQuery.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        /// </summary>
+        public bool? IsBlocked;
+
+        /// <summary>
+        /// If true, get only users that are external
+        /// </summary>
+        public bool? IsExternal;
+
+        /// <summary>
+        /// Exclude the extenral users
```
Comment (louis-z) [inline NGitLab/Models/UserQuery.cs]:
```suggestion
        /// Exclude external users
```

File: NGitLab/Models/UserQuery.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿namespace NGitLab
+{
+    public class UserQuery
+    {
+        /// <summary>
+        /// Specifies how many record per paging (Gitlab supports a maximum of 100 projects and defaults to 20).
+        /// </summary>
+        public int? PerPage;
```
Comment (meziantou) [inline NGitLab/Models/UserQuery.cs]: Idem for other properties

```suggestion
        public int? PerPage { get; set;} 
```

File: NGitLab.Mock/UserCollection.cs [hunk 4]
```diff
@@ -63,5 +63,58 @@ internal IEnumerable<User> SearchByUsername(string username)
         {
             return this.Where(user => string.Equals(user.UserName, username, StringComparison.OrdinalIgnoreCase));
         }
+
+        internal IEnumerable<User> Get(UserQuery query)
+        {
+            var users = this.AsQueryable();
+
+            if (query.IsActive.HasValue && query.IsActive.Value)
```
Comment (louis-z) [inline NGitLab.Mock/UserCollection.cs]: You can make the condition simpler and clearer as follows:
```suggestion
            if (query.IsActive == true)
```

Same thing for other following conditions...

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Models/UserQuery.cs (hunk 3 above)
Comment (Toa741) [reply NGitLab/Models/UserQuery.cs]: Updated

File: NGitLab/Models/UserQuery.cs (hunk 2 above)
Comment (Toa741) [reply NGitLab/Models/UserQuery.cs]: Updated

File: NGitLab.Mock/UserCollection.cs (hunk 4 above)
Comment (Toa741) [reply NGitLab.Mock/UserCollection.cs]: Updated

File: NGitLab/Models/UserQuery.cs (hunk 1 above)
Comment (Toa741) [reply NGitLab/Models/UserQuery.cs]: Updated. I also changed the other queries that had the same summary

**Review by louis-z — APPROVED** [review]

**Pull request merged**